crates/http/src/routes.rs

```diff
@@ -47,37 +51,47 @@ impl Router {
         Ok((Self { routes }, duplicates))
     }
 
-    /// This returns the component ID that should handle the given path, or an error
-    /// if no component matches.
-    ///
-    /// If multiple components could potentially handle the same request based on their
-    /// defined routes, components with matching exact routes take precedence followed
-    /// by matching wildcard patterns with the longest matching prefix.
-    pub(crate) fn route(&self, p: &str) -> Result<&str> {
+    /// Returns the constructed routes.
+    pub fn routes(&self) -> impl Iterator<Item = (&RoutePattern, &String)> {
+        self.routes.iter()
```

Comment:
Probably needs some of this action:
```suggestion
        self.routes.iter().map(|(a, b)| (&a, &b))
```

Reply:
Doesn't seem like that's necessary.

Reply:
Weird!

Reply:
I guess I'm thinking of `HashMap<String, String> -> Iterator<Item = (&str, &str)>`